addons/account_update_tax_tags/wizard/account_update_tax_tags_wizard.py

```diff
@@ -0,0 +1,169 @@
+                SELECT aml.id AS aml_id, rep_tags.account_account_tag_id AS tag_id
+                  FROM account_move_line aml
+                  JOIN account_move move 
+                    ON aml.move_id = move.id AND aml.date >= %(from_date)s
+                  JOIN account_move_line_account_tax_rel amlatr 
+                    ON amlatr.account_move_line_id = aml.id
+             -- Handle possible children_tax_ids
+             LEFT JOIN account_tax_filiation_rel atf 
+                    ON atf.parent_tax = amlatr.account_tax_id
+                  JOIN account_tax tax 
+                    ON tax.id = COALESCE(atf.child_tax, amlatr.account_tax_id)
+                  JOIN account_tax_repartition_line atrl
+                    -- the repartition line to join depends of the move_type
+                    -- also, as this is the base line query, we only join for the base repartition type
+                    ON
+                        atrl.repartition_type = 'base'
+                        AND (
+                            -- invoice type doc
+                            (move.move_type IN ('in_invoice','out_invoice', 'in_receipt', 'out_receipt') AND atrl.invoice_tax_id = tax.id)
+                            OR
+                            -- refund type doc
+                            (move.move_type IN ('in_refund','out_refund') AND atrl.refund_tax_id = tax.id) 
+                            OR
+                            -- entry type doc: depends on the tax type
+                            -- for base line:
+                            -- balance < 0 and sale tax --> invoice
+                            -- balance > 0 and sale tax --> refund
+                            -- balance > 0 and purchase tax --> invoice
+                            -- balance < 0 and purchase tax --> refund
+                            -- impossible to decide for balance at 0 --> invoice by default but it probably doesn't even matter
+                            (
+                                move.move_type = 'entry' 
+                                AND (
+                                    (tax.type_tax_use = 'sale'
+                                    AND (
+                                        aml.balance <= 0 AND atrl.invoice_tax_id = tax.id
+                                        OR aml.balance > 0 AND atrl.refund_tax_id = tax.id 
+                                    ))
+                                    OR
+                                    (tax.type_tax_use = 'purchase'
+                                    AND (
+                                        aml.balance >= 0 AND atrl.invoice_tax_id = tax.id
+                                        OR aml.balance < 0 AND atrl.refund_tax_id = tax.id
+                                    ))
+                                )
+                            ) 
+                        ) 
+             -- LEFT to allow keeping all aml even those without relation. The NULL will symbolize no relation
```

Comment:
To symbolize that there shouldn't be any relation.
If a tax repartition had a tag and then doesn't, the amls that were created with the tag needs to get that tag removed.
If we don't, it could results in wrong reporting.